### PR TITLE
Revert and refactor import criteria functionality

### PR DESCRIPTION
**Pull request by ewelinagr**
[TMT-555](https://jira.thehyve.nl/browse/TMT-555) & [TMT-556](https://jira.thehyve.nl/browse/TMT-556)

TMT-556 is not fully completed - it depends on the variables panel that will be changed in [TMT-631](https://jira.thehyve.nl/browse/TMT-631), so the variables import will be changed and finished accordingly.

### CODE REVIEW

File: src/app/utilities/file-import-helper.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    // reset the input path so that it will take the same file again
+    uploadElm['value'] = '';
+    uploadElm.click();
+  }
+
+  static fileUpload(event, reader: FileReader) {
+    MessageHelper.alert('info', 'File is being processed, waiting for response.');
+    let file = event.target.files[0];
+    reader.readAsText(file);
```
Comment (beaugogh) [inline src/app/utilities/file-import-helper.ts]: also a message telling it's finished?

File: src/app/services/tree-node.service.ts [hunk 2]
```diff
@@ -357,15 +357,15 @@ export class TreeNodeService {
    * @param {string[]} items
    * @param {string[]} paths
    */
-  public convertItemsToPaths(nodes: TreeNode[], items: string[], paths: string[]) {
+  public static convertItemsToPaths(nodes: TreeNode[], items: string[], paths: string[]) {
```
Comment (beaugogh) [inline src/app/services/tree-node.service.ts]: why is this changed to static?

**Comment by beaugogh** [comment]
A general remark: we had issues with these two imports consistently, it would be better to add some integration or e2e tests that also cover edge cases

File: src/app/utilities/file-import-helper.ts (hunk 1 above)
Comment (ewelinagr) [reply src/app/utilities/file-import-helper.ts]: I can add one, but there are already 3 messages showing up, when successfully loading file - 1 on start and 2 when executing restoreCohort. Maybe it would be too much?

**Comment by ewelinagr** [comment]
@beaugogh you're right about the tests, I created an extra ticket: https://jira.thehyve.nl/browse/TMT-636

File: src/app/utilities/file-import-helper.ts (hunk 1 above)
Comment (beaugogh) [reply src/app/utilities/file-import-helper.ts]: I would put up a message at the beginning of the import (inside the importCriteria method of gb-cohort-selection.component.ts) saying it has started, and another message when onLoad is finished, i.e. in handleVariablesFileUploadEvent saying it is finished, and in case error, say there is error (which we already have)

File: src/app/utilities/file-import-helper.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import {MessageHelper} from './message-helper';
+
+export class FileImportHelper {
+
+  static importCriteria(elementId: string, reader: FileReader, isUploadListenerNotAdded: boolean) {
+    let uploadElm = document.getElementById(elementId);
+    if (isUploadListenerNotAdded) {
+      uploadElm.addEventListener('change', (event) => this.fileUpload(event, reader), false);
```
Comment (beaugogh) [inline src/app/utilities/file-import-helper.ts]: `this.fileUpload` -> FileImportHelper.fileUpload
